Source/Shaders/SkyAtmosphereCommon.glsl

```diff
@@ -0,0 +1,264 @@
+        float fLightAngle = dot(lightDirection, samplePoint) / height;
+        float fCameraAngle = dot(ray, samplePoint) / height;
+        float fScatter = (startOffset + depth*(scale(fLightAngle) - scale(fCameraAngle)));
+        vec3 attenuate = exp(-fScatter * (InvWavelength * Kr4PI + Km4PI));
+        frontColor += attenuate * (depth * scaledLength);
+        samplePoint += sampleRay;
+    }
+
+    // Finally, scale the Mie and Rayleigh colors and set up the varying variables for the pixel shader
+    mieColor = frontColor * KmESun;
+    rayleighColor = frontColor * (InvWavelength * KrESun);
+}
+
+vec4 calculateFinalColor(vec3 positionWC, vec3 toCamera, vec3 lightDirection, vec3 mieColor, vec3 rayleighColor)
+{
+    // Extra normalize added for Android
+    float cosAngle = dot(lightDirection, normalize(toCamera)) / length(toCamera);
+    float rayleighPhase = 0.75 * (1.0 + cosAngle * cosAngle);
+    float miePhase = 1.5 * ((1.0 - g2) / (2.0 + g2)) * (1.0 + cosAngle * cosAngle) / pow(1.0 + g2 - 2.0 * g * cosAngle, 1.5);
+
+    vec3 rayleighFinal = rayleighPhase * rayleighColor;
+    vec3 mieFinal = miePhase * mieColor;
+    vec3 rgb = rayleighFinal + mieFinal;
+
+    if (rgb.b > 1000000.0)
+    {
+        // Discard colors that exceed some large number value to prevent against NaN's from the exponent calculation below
+        return vec4(0.0);
+    }
+
+    const float exposure = 2.0;
+    vec3 rgbExposure = vec3(1.0) - exp(-exposure * rgb);
+
+#ifndef HDR
+    rgb = rgbExposure;
+#endif
+
+#ifdef COLOR_CORRECT
+    // Convert rgb color to hsb
+    vec3 hsb = czm_RGBToHSB(rgb);
+    // Perform hsb shift
+    hsb.x += u_hsbShift.x; // hue
+    hsb.y = clamp(hsb.y + u_hsbShift.y, 0.0, 1.0); // saturation
+    hsb.z = hsb.z > czm_epsilon7 ? hsb.z + u_hsbShift.z : 0.0; // brightness
+    // Convert shifted hsb back to rgb
+    rgb = czm_HSBToRGB(hsb);
+#endif
+
+    float outerRadius = u_radiiAndDynamicAtmosphereColorAndInverseScale.x;
+    float innerRadius = u_radiiAndDynamicAtmosphereColorAndInverseScale.y;
+    float lightEnum = u_radiiAndDynamicAtmosphereColorAndInverseScale.z;
+
+    // Alter alpha based on how close the viewer is to the ground (1.0 = on ground, 0.0 = at edge of atmosphere)
+    float atmosphereAlpha = clamp((outerRadius - length(positionWC)) / (outerRadius - innerRadius), 0.0, 1.0);
+
+    // Alter alpha based on time of day (0.0 = night , 1.0 = day)
+    float nightAlpha = (lightEnum != 0.0) ? clamp(dot(normalize(positionWC), lightDirection), 0.0, 1.0) : 1.0;
+    atmosphereAlpha *= pow(nightAlpha, 0.5);
+
+    vec4 finalColor = vec4(rgb, mix(clamp(rgbExposure.b, 0.0, 1.0), 1.0, atmosphereAlpha) * smoothstep(0.0, 1.0, czm_morphTime));
+
+    float strength = mieFinal.b * rayleighColor.b * 0.05;
+    float minStrength = 0.0;
+    float maxStrength = 1.0;
+    strength = clamp(strength, minStrength, maxStrength);
+    float alpha = (strength - minStrength) / maxStrength;
+    finalColor.a = (1.0 - alpha);
```

Comment:
Still getting
<img width="797" alt="Screen Shot 2020-05-20 at 11 18 49 AM" src="https://user-images.githubusercontent.com/1328450/82464344-f765d880-9a8b-11ea-98ba-112a9aa99151.png">

in vertex mode

Reply:
Debugging the alpha value:
<img width="786" alt="Screen Shot 2020-05-20 at 11 22 35 AM" src="https://user-images.githubusercontent.com/1328450/82464700-63e0d780-9a8c-11ea-8bb0-6dea942935b3.png">